server/src/main/java/org/folio/reservoir/server/IngestWriteStream.java

```diff
@@ -0,0 +1,136 @@
+package org.folio.reservoir.server;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.streams.WriteStream;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.folio.reservoir.module.impl.ModuleJsonPath;
+import org.folio.reservoir.util.SourceId;
+
+public class IngestWriteStream implements WriteStream<JsonObject> {
```

Comment:
Let's commit as is but eventually we will need a util to "wrap" MarcInJson as  "payload" and handle "delete" and MFHD in one place.

Reply:
ok